apps/contributions/models.py

```
@@ -764,6 +772,28 @@ def send_recurring_contribution_email_reminder(self, next_charge_date: datetime.
             next_charge_date,
         )
 
+    def set_metadata_field(self, key: str, value: Any) -> None:
+        """Set a field in contribution_metadata, ensuring that the result is valid according to its schema version.
+
+        If an invalid key or value is set, this will raise an InvalidMetadataError exception and contribution_metadata will not be changed.
+        This doesn't make any changes in Stripe.
+        """
+        if key == "schema_version":
+            raise InvalidMetadataError("Schema version may not be changed")
```

When `InvalidMetadataError `is contextlib.suppressed (below), I assume Sentry never sees it? Thus there is no log of it ever happening.

Do any of error these conditions warrant logging? Perhaps just debug?  And if so, log here? or let the caller decide to log or not (by catching logging and possibly ignoring exception)?

---

My approach here is to let the caller decide, but could re-evaluate if it ends up being a common pattern.